README.md

```diff
@@ -11,195 +11,174 @@ A TypeScript implementation of the C# library [CSharpFunctionalExtensions](https
+} from 'typescript-functional-extensions';
 ```
 
-```typescript
-const maybe = Maybe.tryFirst(['apple', 'banana'], (fruit) => fruit.length > 6);
-
-console.log(maybe.getValueOrThrow()); // throws Error 'No value'
-```
-
-#### TryLast
-
-```typescript
-const maybe = Maybe.tryLast(
-  ['apple', 'banana', 'mango'],
-  (fruit) => fruit.length === 5
-);
-
-console.log(maybe.getValueOrThrow()); // 'mango'
-```
-
-```typescript
-const maybe = Maybe.tryLast(
-  ['apple', 'banana', 'mango'],
-  (fruit) => fruit === 'pear'
-);
+## Monads
 
-console.log(maybe.getValueOrThrow()); // throws Error 'No value'
-```
+Below are the monads included in this package and examples of their use.
 
-#### Map
+More examples of all monads and their methods can be found in the library unit tests or in the dedicated documentation files for each type.
 
-```typescript
-const maybe = Maybe.some({ type: 'fruit', name: 'apple' })
-  .map(({ type }) => ({ type, name: 'banana' }))
-  .map((food) => food.name)
-  .map((name) => name.length);
+### Maybe
 
-console.log(maybe.getValueOrThrow()); // 6
-```
+`Maybe` represents a value that might or might not exist. You can use it to declaratively describe a process (series of steps) without having to check if there is a value present.
 
 ```typescript
-type Food = {
-  type: string;
-  name: string;
+type Employee = {
+  email: string;
+  firstName: string;
+  lastName: string;
+  manager: Employee | undefined;
 };
 
-const maybe = Maybe.none<Food>()
-  .map(({ type }) => ({ type, name: 'banana' }))
-  .map((food) => food.name)
-  .map((name) => name.length);
-
-console.log(maybe.getValueOrThrow()); // throws Error 'No value'
-```
-
-#### Match
-
-```typescript
-const maybe = Maybe.some({ type: 'fruit', name: 'apple' })
-  .map(({ type }) => ({ type, name: 'banana' }))
-  .map((food) => food.name)
-  .map((name) => name.length)
-  .match({
-    some: (number) => console.log(number),
-    none: () => console.log('None!'),
-  }); // 6
-```
-
-```typescript
-type Food = {
-  type: string;
-  name: string;
-};
+function yourBusinessProcess(): Employee[] {
+  // ...
+}
 
-const maybe = Maybe.none<Food>()
-  .map(({ type }) => ({ type, name: 'banana' }))
-  .map((food) => food.name)
-  .map((name) => name.length)
+const employees = yourBusinessProcess();
+
+Maybe.tryFirst(employees)
+  .tap(({ firstName, lastName, email }) =>
+    console.log(`Found Employee: ${firstName} ${lastName}, ${email}`))
+  .bind(employee =>
+    Maybe.from(employee.manager)
+      .or({
+        email: 'supervisor@business.com',
+        firstName: 'Company',
+        lastName: 'Supervisor',
+        manager: undefined
+      })
+      .map(manager => ({ manager, employee }))
+  )
   .match({
-    some: (number) => console.log(number),
-    none: () => console.log('None!'),
-  }); // None!
+    some(attendees => scheduleMeeting(attendees.manager, attendees.employee)),
+    none(() => console.log(`The business process did not return any employees`))
+  });
```

Comment:
This "real world" example is great for showing how this (and the other monads) might work in application code. Big improvement over the previous examples! 👍

Reply:
Thanks! I agree.

For someone not familiar with these ideas, the use-case is likely a bit too abstract by just reading the description of the library so hopefully these code-in-action examples give people an idea of what's possible.